src/Mono.Android/Microsoft.Android.Runtime/ManagedValueManager.cs

```diff
@@ -20,53 +22,69 @@ class ManagedValueManager : JniRuntime.JniValueManager
 {
 	const DynamicallyAccessedMemberTypes Constructors = DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors;
 
-	Dictionary<int, List<IJavaPeerable>>?   RegisteredInstances = new Dictionary<int, List<IJavaPeerable>>();
+	Dictionary<int, List<ReferenceTrackingHandle>>? RegisteredInstances = new ();
```

Comment:
This code was like this before, but:
* Does anything set `RegisteredInstances` to null?
* If not, can we make it `readonly` and non-nullable?

There are several checks in here like:
```csharp
if (RegisteredInstances == null)
	throw new ObjectDisposedException (nameof (ManagedValueManager));
```

Should those actually be checking if `_context` is null?

Reply:
I agree, we can drop the `?` + remove all the null checks.

I am not sure if this class can even be disposed. There is no way to _deregister_ from `JavaMarshal` so it will keep calling the UCO methods. It seems like `Dispose` on this class should crash the whole app 🤔 Is `JniRuntime.Dispose()` part of graceful shutdown of the app? or is that ever called? I don't see how else the VM would be disposed.

Reply:
`Dispose()` is probably never called on Android, it might be called on certain desktop unit tests:
* https://github.com/dotnet/java-interop/blob/5c48d83c922ca30026110e416b9a6570dbcbbd4c/tests/Java.Interop-Tests/Java.Interop/JniRuntimeTest.cs#L23-L38